### PR TITLE
Fix/create customer in draft order

### PR DESCRIPTION
**Pull request by sebastiannicolajsen**
### What
- Add update statement after creating a cart from `draft-order`. This is required to actually create the user (if this is a new user)

### Why?
- It might be necessary to use new emails when creating an `order` from a `draft-order`. However, doing so right now results in an error once the user tries to mark the `order` as paid, since the new user was never instantiated.

### How?
- Fetch the customer during creation of a cart

### CODE REVIEW

**Review by srindom — COMMENTED** [review]
what is the idea here? :)

**Comment by sebastiannicolajsen** [comment]
@srindom updated description

**Comment by srindom** [comment]
Can we make the customer part of the `create` method instead of having to make two calls to the cart service :)

File: packages/medusa/src/services/cart.js [hunk 1]
```diff
@@ -306,6 +306,16 @@ class CartService extends BaseService {
 
       const regCountries = region.countries.map(({ iso_2 }) => iso_2)
 
+      if (data.customer_id) {
+        const customer = await this.customerService_
+          .withTransaction(this.transactionManager_)
```
Comment (srindom) [inline packages/medusa/src/services/cart.js:311]: should use `manager` instead of `this.transactionManager_`

**Review by srindom — CHANGES_REQUESTED** [review]
Can you add a unit and integration test for this please :)

File: packages/medusa/src/services/cart.js [hunk 2]
```diff
@@ -760,12 +769,29 @@ class CartService extends BaseService {
...
+        MedusaError.Types.INVALID_DATA,
+        "The email is not valid"
+      )
+    }
+
+    const customer = await this.customerService_
+      .withTransaction(this.transactionManager_)
+      .create({ email })
+
+    return customer
```
Comment (srindom) [inline packages/medusa/src/services/cart.js:794]: This is not used

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa-payment-manual/services/manual-payment.js [hunk 3]
```diff
@@ -0,0 +1,364 @@
+"use strict";
```
Comment (srindom) [inline packages/medusa-payment-manual/services/manual-payment.js:1]: Should be `.gitignore`d

**Review by srindom — DISMISSED** [review]

**Review by srindom — APPROVED** [review]

**Review by srindom — APPROVED** [review]

**Pull request merged**